src/main/java/org/springframework/integration/aws/inbound/S3InboundStreamingMessageSource.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.integration.aws.inbound;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Comparator;
+import java.util.List;
+
+import org.springframework.integration.aws.support.S3FileInfo;
+import org.springframework.integration.file.remote.AbstractFileInfo;
+import org.springframework.integration.file.remote.AbstractRemoteFileStreamingMessageSource;
+import org.springframework.integration.file.remote.RemoteFileTemplate;
+
+import com.amazonaws.services.s3.model.S3ObjectSummary;
+
+/**
+ * A {@link AbstractRemoteFileStreamingMessageSource} implementation for the Amazon S3.
+ *
+ * @author Christian Tzolov
```

Comment:
Yes, fully similar to `(S)FtpFileInfo`.
I don't see reason to hide our classes from end-users.
They might be useful somehow.
And that is end-user responsibility to treat then on his own.

And yes: new test classes must be be marked with `@since` as well.